autobahn/wamp_subscribe_options.ipp

```diff
@@ -0,0 +1,107 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) Tavendo GmbH
+//
+// Boost Software License - Version 1.0 - August 17th, 2003
+//
+// Permission is hereby granted, free of charge, to any person or organization
+// obtaining a copy of the software and accompanying documentation covered by
+// this license (the "Software") to use, reproduce, display, distribute,
+// execute, and transmit the Software, and to prepare derivative works of the
+// Software, and to permit third-parties to whom the Software is furnished to
+// do so, all subject to the following:
+//
+// The copyright notices in the Software and this entire statement, including
+// the above license grant, this restriction and the following disclaimer,
+// must be included in all copies of the Software, in whole or in part, and
+// all derivative works of the Software, unless such copies or derivative
+// works are solely in the form of machine-executable object code generated by
+// a source language processor.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT
+// SHALL THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING THE SOFTWARE BE LIABLE
+// FOR ANY DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <msgpack.hpp>
+#include <string>
+#include <unordered_map>
+
+namespace autobahn {
+
+inline wamp_subscribe_options::wamp_subscribe_options()
+    : m_match("exact")
+{
+}
+
+inline const std::string& wamp_subscribe_options::match() const
+{
+    return m_match;
+}
+
+inline void wamp_subscribe_options::set_match(const std::string& match)
+{
+    m_match = match;
+}
+
+} // namespace autobahn
+
+namespace msgpack {
+MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS) {
+namespace adaptor {
+
+template<>
+struct convert<autobahn::wamp_subscribe_options>
+{
+    msgpack::object const& operator()(
+            msgpack::object const& object,
+            autobahn::wamp_subscribe_options& options) const
+    {
+        std::unordered_map<std::string, msgpack::object> options_map;
+        object >> options_map;
+
+        return object;
+    }
+};
+
+template<>
+struct pack<autobahn::wamp_subscribe_options>
+{
+    template <typename Stream>
+    msgpack::packer<Stream>& operator()(
+            msgpack::packer<Stream>& packer,
+            autobahn::wamp_subscribe_options const& options) const
+    {
+        std::unordered_map<std::string, std::string> options_map;
```

Comment:
As a side note, over time I've grown wary of unwarranted use of hashmaps. The extra hashing takes time and extra allocations that might pay off for heavily read maps with thousands of elements, but is worse off with few elements and more writing than retrieving... such as an options map. I think std::map would be more suitable for this kind of use.